### PR TITLE
CDD-1505 Added the connector to send request to Notification service,…

### PR DESCRIPTION
**Pull request by usmanshabbir**
… refactored some tests

### CODE REVIEW

File: app/uk/gov/hmrc/customs/declaration/connectors/CustomsNotificationConnector.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      (ACCEPT -> MimeTypes.XML),
+      (AUTHORIZATION -> s"Basic $basicAuthToken"))
+
+
+    http.POSTString(
+      serviceUrl,
+      XMLHeader + notification.payload.toString(),
+      headers.toSeq
+    ) map (x => ()) recover {
+      case e: Throwable =>
```
Comment (googley42) [inline app/uk/gov/hmrc/customs/declaration/connectors/CustomsNotificationConnector.scala]: Question - is it a requirement to swallow exception here?

File: test/acceptance/UploadedFileUpscanNotificationSpec.scala [hunk 2]
```diff
@@ -18,16 +18,14 @@ package acceptance
 
 import java.util.UUID
```
Comment (googley42) [inline test/acceptance/UploadedFileUpscanNotificationSpec.scala]: unused import `import org.scalatest.concurrent.{Eventually, IntegrationPatience}`

File: test/unit/connectors/CustomsNotificationConnectorSpec.scala [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      await(connector.send(customsNotification))
+
+      verify(mockHttpClient).POSTString(
+        ameq(expectedUrl),
+        ameq(s"""<?xml version="1.0" encoding="UTF-8" standalone="yes"?>""" + customsNotification.payload.toString()),
+        ameq(expectedHeaders.toSeq)
+      )(any(), any(), any())
+    }
+
+    "log the error in case request to notification service fails" in {
```
Comment (googley42) [inline test/unit/connectors/CustomsNotificationConnectorSpec.scala]: we have constant in [emulatedServiceFailure](https://github.com/hmrc/customs-declarations/blob/master/test/util/TestData.scala#L57) for this

File: app/uk/gov/hmrc/customs/declaration/connectors/CustomsNotificationConnector.scala (hunk 1 above)
Comment (usmanshabbir) [reply app/uk/gov/hmrc/customs/declaration/connectors/CustomsNotificationConnector.scala]: for now yes, we cant recover from this exception.

File: app/uk/gov/hmrc/customs/declaration/connectors/CustomsNotificationConnector.scala [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      ("X-Conversation-ID" -> notification.conversationId.toString),
+      (CONTENT_TYPE -> s"${MimeTypes.XML}; charset=UTF-8"),
+      (ACCEPT -> MimeTypes.XML),
+      (AUTHORIZATION -> s"Basic $basicAuthToken"))
+
+
+    http.POSTString(
+      serviceUrl,
+      XMLHeader + notification.payload.toString(),
+      headers.toSeq
```
[the remark this answers is not in the document]
Comment (usmanshabbir) [reply app/uk/gov/hmrc/customs/declaration/connectors/CustomsNotificationConnector.scala]: 👍

File: test/acceptance/UploadedFileUpscanNotificationSpec.scala (hunk 2 above)
Comment (usmanshabbir) [reply test/acceptance/UploadedFileUpscanNotificationSpec.scala]: 👍

File: test/unit/connectors/CustomsNotificationConnectorSpec.scala (hunk 3 above)
Comment (usmanshabbir) [reply test/unit/connectors/CustomsNotificationConnectorSpec.scala]: that is unsupported operation exception, I wanted to make sure that it works for any runtime exception hence the parent RuntimeException is used.

File: app/uk/gov/hmrc/customs/declaration/connectors/CustomsNotificationConnector.scala [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+@Singleton
+class CustomsNotificationConnector @Inject()(http: HttpClient,
+                                             logger: CdsLogger,
+                                             configs: ServicesConfig) {
+
+
```
[the remark this answers is not in the document]
Comment (usmanshabbir) [reply app/uk/gov/hmrc/customs/declaration/connectors/CustomsNotificationConnector.scala]: I don't think the code is messy, however, I do agree that we should fail the service on startup and the service should not return 200 for health endpoint when all the required configs are not provided.

**Pull request merged**